benchmarks/src/main/java/zipkin2/it/ServerIntegratedBenchmark.java

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright 2015-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.it;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.Closer;
+import com.linecorp.armeria.client.HttpClient;
+import io.netty.handler.codec.http.QueryStringEncoder;
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.utility.MountableFile;
+
+class ServerIntegratedBenchmark {
+
+  static final Logger LOGGER = LoggerFactory.getLogger(ServerIntegratedBenchmark.class);
+
+  static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  @Nullable static final String RELEASED_ZIPKIN_VERSION = System.getenv("ZIPKIN_VERSION");
+
+  static final boolean WAIT_AFTER_BENCHMARK = "true".equals(System.getenv("ZIPKIN_BENCHMARK_WAIT"));
+
+  Closer closer;
+
+  @BeforeEach void setUp() {
+    closer = Closer.create();
+  }
+
+  @AfterEach void tearDown() throws Exception {
+    closer.close();
+  }
+
+  @Test void inMemory() throws Exception {
+    runBenchmark(null);
+  }
+
+  @Test void elasticsearch() throws Exception {
+    GenericContainer<?> elasticsearch = new GenericContainer<>("openzipkin/zipkin-elasticsearch7")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("elasticsearch")
+      .withLabel("name", "elasticsearch")
+      .withExposedPorts(9200)
+      .waitingFor(new HttpWaitStrategy().forPath("/_cluster/health"));
+    startContainer(elasticsearch);
+    runBenchmark(elasticsearch);
+  }
+
+  @Test void cassandra() throws Exception {
+    GenericContainer<?> cassandra = new GenericContainer<>("openzipkin/zipkin-cassandra")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("cassandra3")
```

Comment:
note: we use "cassandra" in docker compose because this works for any of our cassandra storage types. prefer parity with the docker-compose files when naming etc.